internal/backend/deployments/deployments.go

```diff
@@ -144,6 +154,27 @@ func drain(ctx context.Context, tx db.DB, id sdktypes.DeploymentID) error {
 	return nil
 }
 
+func updateDeploymentState(ctx context.Context, db db.DB, id sdktypes.DeploymentID, state sdktypes.DeploymentState) error {
+	oldState, err := db.UpdateDeploymentState(ctx, id, state)
+	if err != nil {
+		return err
+	}
+
+	updateStateCounter := func(state sdktypes.DeploymentState, val int64) {
+		switch state {
+		case sdktypes.DeploymentStateActive:
+			deploymentsActiveCounter.Add(ctx, val)
+		case sdktypes.DeploymentStateDraining:
+			deploymentsDrainingCounter.Add(ctx, val)
+		case sdktypes.DeploymentStateInactive:
+			deploymentsInactiveCounter.Add(ctx, val)
+		}
+	}
+	updateStateCounter(state, 1)
+	updateStateCounter(oldState, -1)
```

Comment:
you can't update a counter with -1
maybe you can technically
but a counter should represent and ever increasing value

a gauge or an up/down counter should represent a varying value

Reply:
do you want me to change name to represent it?

Reply:
i guess otel supports that?